### PR TITLE
Geometrical info

### PR DESCRIPTION
**Pull request by rijobro**
Trying to get this PR going again. Here are the changes I made:

1. In keeping with #241, I reverted your most recent commit, which was the addition of ublas. I assumed that everything was working fine before that commit, in which case everything should still be OK.
2. Merge in master, resolve conflicts (two tiny ones on CMake level).
3. Use the `sirf` namespace. Also change downcast from `PETImageData` to `STIRImageData` according to new image hierarchy (#226).
4. Put VoxelisedGeometricalInfo into ImageData
    1. `ImageData` stores a shared pointer of `GeomInfo`
    2. This `GeomInfo` is created the first time it is requested (with `get_geom_info`)
    3. Abstract method `set_up_geom_info`.
    4. Moved implementation from `get_patient_coord_geometrical_info` to `set_up_geom_info`
    5. Small effort at filling in `GeomInfo` for gadgetron images

### CODE REVIEW

**Comment by rijobro** [comment]
three more commits:
1. Merge in master (`image_data.h` -> `ImageData.h`)
2. `geometrical_info` -> `GeometricalInfo` (in keeping with https://github.com/CCPPETMR/SIRF/issues/248)
3. Initially, I was going to initalise the GeomInfo the first time `get_geom_info` was used. Unfortunately, I can't do that as I need the `get_geom_info` to be `const`. Instead, <b>every</b> constructor will have to call the `set_up_geom_info()` method.

**Comment by rijobro** [comment]
@KrisThielemans , @evgueni-ovtchinnikov : any chance you could look this PR over? I can't assign reviewers...

**Comment by evgueni-ovtchinnikov** [comment]
looks ok to me

**Review by evgueni-ovtchinnikov — APPROVED** [review]

File: src/xSTIR/cSTIR/stir_data_containers.h [hunk 1]
```diff
@@ -582,57 +582,47 @@ namespace sirf {
...
 		STIRImageData(std::string filename)
 		{
 			_data = stir::read_from_file<Image3DF>(filename);
+            this->set_up_geom_info();
 		}
 		STIRImageData* same_image_data()
 		{
 			STIRImageData* ptr_image = new STIRImageData;
 			ptr_image->_data.reset(_data->get_empty_copy());
+            this->set_up_geom_info();
```
Comment (KrisThielemans) [inline src/xSTIR/cSTIR/stir_data_containers.h:625]: doesn't make sense here. it's not modifying `*this` at all but creating a copy. In fact, this should be a `const` method

**Review by KrisThielemans — APPROVED** [review]
I have one small comment. However, I suggest @ashgillman merges (not rebases) this now to have only 1 PR to think about.... Possibly then address my comment (if it's valid)

**Pull request merged**